doc/pinkybar.1
.\"t
.\" Automatically generated by Pandoc 1.19.2.1
.\"
.TH "pinkybar" "1" "Oktoberfest 23, 2016" "manual" ""
.hy
.SH NAME
.PP
This man page is converted from markdown, and some information might not
look good and/or have been stripped during the automated process.
.SH SYNOPSIS
.PP
pinkybar [\f[I]option\f[]]
.SH DESCRIPTION
.PP
Gather some system information and show it in this statusbar program,
not tied to any Window Manager, terminal multiplexer, etc.
.PP
Please note that the program won't detect fans connected via molex
connetor(s) or external fan controller.
Also I have not tested it with fan splitter(s) either.
.PP
The program is smart enough to detect whether some of your fan(s) blades
are spinning, or the particular fan have been removed.
Hold down some of your fan blades and you'll see that the program won't
include this fan and it's RPM, release the blades and you'll see the fan
and it's RPM in the statusbar.
Try simulating real fan hardware failure by holding down all system fan
blades and watch what the program will show you, just try not to slice
your cheesy fingers open in the process.
.PP
You can extend pinky\-bar with your own crafted
perl/python/ruby/lua/R/assembly/lisp/ocaml/rust/go/c++/slang/tcl script.
.PP
If you compile your kernel from source code make sure to include your
cpu and motherboard sensors as \f[B]modules\f[] and not inlined.
.PP
\f[B]Just an example if you use BSD \- acpi/aibs, coretemp/amdtemp.\f[]
.PP
[IMAGE: image (img/cpu-temp.png)]
.PP
[IMAGE: image (img/mobo-temp.png)]
.PP
   *   *   *   *   *
.SS Program options
.PP
The order of supplied options will dictate how, where and what system
information to be shown.
.PP
.TS
tab(@);
lw(10.7n) lw(10.0n) lw(49.3n).
T{
short option
T}@T{
long option
T}@T{
Descrtiption
T}
_
T{
\-M
T}@T{
\[en]mpd
T}@T{
The song filename
T}
T{
\-W
T}@T{
\[en]mpdtrack
T}@T{
The song track name (not available in cmus)
T}
T{
\-x
T}@T{
\[en]mpdartist
T}@T{
The song artist(s) name(s)
T}
T{
\-X
T}@T{
\[en]mpdtitle
T}@T{
The song title
T}
T{
\-y
T}@T{
\[en]mpdalbum
T}@T{
The song album name
T}
T{
\-Y
T}@T{
\[en]mpddate
T}@T{
The song date
T}
T{
\-c
T}@T{
\[en]cpu
T}@T{
The current cpu load (summed up all cores/threads)
T}
T{
\-L
T}@T{
\[en]coresload
T}@T{
Show the load regarding each individual cpu core/thread
T}
T{
\-T
T}@T{
\[en]cputemp
T}@T{
The current cpu temperature
T}
T{
\-C
T}@T{
\[en]cpuspeed
T}@T{
Show your maximum cpu clock speed in MHz, regardless of the used
governor.
Uses assembly.
T}
T{
\-I
T}@T{
\[en]cpuinfo
T}@T{
Detect your CPU vendor, stepping, family, clflush, l1/l2 cache and line
size, physical cores, physical and virtual bits.
Uses assembly.
T}
T{
\-r
T}@T{
\[en]ramperc
T}@T{
The used ram in percentage
T}
T{
\-J
T}@T{
\[en]ramtotal
T}@T{
The total ram
T}
T{
\-K
T}@T{
\[en]ramfree
T}@T{
The free ram
T}
T{
\-l
T}@T{
\[en]ramshared
T}@T{
The shared ram
T}
T{
\-o
T}@T{
\[en]rambuffer
T}@T{
The buffer ram (not available in OpenBSD)
T}
T{
\-s
T}@T{
\[en]driveperc
T}@T{
The used drive storage in percentage
T}
T{
\-n
T}@T{
\[en]drivetotal
T}@T{
The total drive storage
T}
T{
\-N
T}@T{
\[en]drivefree
T}@T{
The free drive storage
T}
T{
\-O
T}@T{
\[en]driveavail
T}@T{
The available drive storage (total \- used)
T}
T{
T}@T{
\[en]drivetemp
T}@T{
Read the drive temperature from S.M.A.R.T
T}
T{
\-g
T}@T{
\[en]battery
T}@T{
The remaining battery charge
T}
T{
\-z
T}@T{
\[en]dvdstr
T}@T{
The vendor and model name of your cdrom/dvdrom
T}
T{
\-S
T}@T{
\[en]statio
T}@T{
Read and written MBs to the drive so far [argument \- sda]
T}
T{
\-p
T}@T{
\[en]packages
T}@T{
The number of installed packages
T}
T{
\-P
T}@T{
\[en]kernsys
T}@T{
The kernel name
T}
T{
T}@T{
\[en]kernode
T}@T{
The network node hostname
T}
T{
\-Q
T}@T{
\[en]kernrel
T}@T{
The kernel release
T}
T{
\-R
T}@T{
\[en]kernver
T}@T{
The kernel version
T}
T{
\-u
T}@T{
\[en]kernarch
T}@T{
The machine architecture
T}
T{
\-k
T}@T{
\[en]kernel
T}@T{
Combined kernel name and version
T}
T{
T}@T{
\[en]perl
T}@T{
Extend pinkybar with your scripts written in perl, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]python
T}@T{
Extend pinkybar with your scripts written in python, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]ruby
T}@T{
Extend pinkybar with your scripts written in ruby, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]lua
T}@T{
Extend pinkybar with your scripts written in lua, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]R
T}@T{
Extend pinkybar with your scripts written in R, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]asm
T}@T{
Extend pinkybar with assembly, learn more from the Opt\-in section.
T}
T{
T}@T{
\[en]lisp
T}@T{
Extend pinkybar with your scripts written in lisp, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]ocaml
T}@T{
Extend pinkybar with your scripts written in ocaml, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]rust
T}@T{
Extend pinkybar with your scripts written in rust, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]go
T}@T{
Extend pinkybar with your scripts written in go, learn more from the
Opt\-in section.
T}
T{
T}@T{
\[en]cpp
T}@T{
Extend pinkybar with your cpp, learn more from the Opt\-in section.
T}
T{
T}@T{
\[en]slang
T}@T{
Extend pinkybar with your slang, learn more from the Opt\-in section.
T}
T{
T}@T{
\[en]tcl
T}@T{
Extend pinkybar with your tcl, learn more from the Opt\-in section.
T}
T{
\-q
T}@T{
\[en]weather
T}@T{
Show the temperature outside [argument \- London,uk]
T}
T{
\-U
T}@T{
\[en]uptime
T}@T{
The system uptime
T}
T{
\-w
T}@T{
\[en]loadavg
T}@T{
The system average load for past 1, 5 and 15 minutes
T}
T{
\-v
T}@T{
\[en]voltage
T}@T{
The system voltage
T}
T{
\-f
T}@T{
\[en]fans
T}@T{
All system fans and their speed in RPM
T}
T{
\-m
T}@T{
\[en]mobo
T}@T{
Show the motherboard name and vendor
T}
T{
\-d
T}@T{
\[en]mobotemp
T}@T{
The motherboard temperature
T}
T{
\-V
T}@T{
\[en]volume
T}@T{
The sound volume level
T}
T{
\-t
T}@T{
\[en]time
T}@T{
The current time
T}
T{
\-a
T}@T{
\[en]ipaddr
T}@T{
The local ip address [argument \- eth0]
T}
T{
\-b
T}@T{
\[en]bandwidth
T}@T{
The consumed internet bandwidth so far [argument \- eth0]
T}
T{
\-i
T}@T{
\[en]iface
T}@T{
The current download and upload speed [argument \- eth0]
T}
T{
\-A
T}@T{
\[en]ipmac
T}@T{
The NIC mac address [argument \- eth0]
T}
T{
\-B
T}@T{
\[en]ipmask
T}@T{
The NIC subnet mask [argument \- eth0]
T}
T{
\-D
T}@T{
\[en]ipcast
T}@T{
The NIC broadcast address [argument \- eth0]
T}
T{
\-E
T}@T{
\[en]iplookup
T}@T{
Mini website IP lookup [website argument \- google.com]
T}
.TE
.PP
Be aware of the options that mention \f[B]Uses assembly\f[] are tested
only on AMD and Intel CPUs (starting from pentium 4 onwards).
.PP
The following options are available only in Linux:
.PP
.TS
tab(@);
lw(10.7n) lw(10.0n) lw(49.3n).
T{
short option
T}@T{
long option
T}@T{
Descrtiption
T}
_
T{
\-F
T}@T{
\[en]drivemodel
T}@T{
The vendor name of your drive [argument \- sda]
T}
T{
\-G
T}@T{
\[en]nicinfo
T}@T{
The NIC vendor and model [argument \- eth0]
T}
T{
T}@T{
\[en]nicdrv
T}@T{
The NIC driver [argument \- eth0]
T}
T{
\-H
T}@T{
\[en]nicver
T}@T{
The NIC version [argument \- eth0]
T}
T{
\-e
T}@T{
\[en]iplink
T}@T{
The NIC link speed (useful for wireless/wifi) [argument \- eth0]
T}
T{
\-j
T}@T{
\[en]nicfw
T}@T{
The NIC firmware [argument \- eth0]
T}
T{
\-h
T}@T{
\[en]wifiname
T}@T{
The name of currently connected wifi/wireless network [argument \-
wlan0]
T}
.TE
.PP
The following options are available only to FreeBSD and OpenBSD:
.PP
.TS
tab(@);
lw(10.7n) lw(10.0n) lw(49.3n).
T{
short option
T}@T{
long option
T}@T{
Descrtiption
T}
_
T{
\-j
T}@T{
\[en]nicgw
T}@T{
The NIC gateway address [argument \- re0]
T}
T{
\-Z
T}@T{
\[en]swapused
T}@T{
The used drive swap in MB
T}
T{
\-F
T}@T{
\[en]swaperc
T}@T{
The used drive swap in percentage
T}
T{
\-h
T}@T{
\[en]swaptotal
T}@T{
The total drive swap
T}
T{
\-H
T}@T{
\[en]swapavail
T}@T{
The available drive swap (total \- used)
T}
T{
T}@T{
\[en]wifiname
T}@T{
The name of currently connected wifi/wireless network [argument \-
wlan0]
T}
.TE
.PP
The following options are available only in OpenBSD:
.PP
.TS
tab(@);
l l l.
T{
short option
T}@T{
long option
T}@T{
Descrtiption
T}
_
T{
\-l
T}@T{
\[en]ramused
T}@T{
The used ram in MB
T}
.TE
.PP
   *   *   *   *   *
.SS GNU Build System (configure) options
.PP
Before the source code is passed to the compiler, you can enable/disable
the following \f[B]configure\f[] options that will increase/reduce the
number of dependencies required to compile the program.
.PP
It's up to you to decide which features suit you best.
.PP
.TS
tab(@);
lw(9.0n) lw(11.7n) lw(49.3n).
T{
To include
T}@T{
Not to include
T}@T{
Descrtiption
T}
_
T{
\[en]with\-x11
T}@T{
\[en]without\-x11
T}@T{
Enable it if you are using dwm.
T}
T{
\[en]with\-alsa
T}@T{
\[en]without\-alsa
T}@T{
To get the sound volume level.
T}
T{
\[en]with\-oss
T}@T{
\[en]without\-oss
T}@T{
To get the sound volume level in *BSD.
T}
T{
\[en]with\-net
T}@T{
\[en]without\-net
T}@T{
Enable the internet related options.
T}
T{
\[en]with\-libnl
T}@T{
\[en]without\-libnl
T}@T{
Enable the wifi related options regarding chipsets supporting the
cfg80211/mac80211 modules (linux only).
T}
T{
\[en]with\-pci
T}@T{
\[en]without\-pci
T}@T{
To get the NIC vendor and model in linux
T}
T{
\[en]with\-dvd
T}@T{
\[en]without\-dvd
T}@T{
To get the cdrom/dvdrom vendor and model
T}
T{
\[en]with\-sensors
T}@T{
\[en]without\-sensors
T}@T{
Alternative way to get data from the sensors (linux only)
T}
T{
\[en]with\-apm
T}@T{
\[en]without\-apm
T}@T{
APM power and resource management for laptops (FreeBSD only)
T}
T{
\[en]with\-ncurses
T}@T{
\[en]without\-ncurses
T}@T{
Output the data to the terminal using the ncurses library, can be
colorized
T}
T{
\[en]with\-perl
T}@T{
\[en]without\-perl
T}@T{
Extend pinkybar with your own crafted scripts written in perl
T}
T{
\[en]with\-lua
T}@T{
\[en]without\-lua
T}@T{
Extend pinkybar with your own crafted scripts written in lua
T}
T{
\[en]with\-ruby
T}@T{
\[en]without\-ruby
T}@T{
Extend pinkybar with your own crafted scripts written in ruby
T}
T{
\[en]with\-python2
T}@T{
\[en]without\-python2
T}@T{
Extend pinkybar with your own crafted scripts written in python2
T}
T{
\[en]with\-python3
T}@T{
\[en]without\-python3
T}@T{
Extend pinkybar with your own crafted scripts written in python3
T}
T{
\[en]with\-r
T}@T{
\[en]without\-r
T}@T{
Extend pinkybar with your own crafted scripts written in R
T}
T{
\[en]with\-assembly
T}@T{
\[en]without\-assembly
T}@T{
Extend pinkybar with assembly
T}
T{
\[en]with\-lisp
T}@T{
\[en]without\-lisp
T}@T{
Extend pinkybar with your own crafted scripts written in lisp
T}
T{
\[en]with\-ocaml
T}@T{
\[en]without\-ocaml
T}@T{
Extend pinkybar with your own crafted scripts written in ocaml
T}
T{
\[en]with\-rust
T}@T{
\[en]without\-rust
T}@T{
Extend pinkybar with your own crafted scripts written in rust
T}
T{
\[en]with\-go
T}@T{
\[en]without\-go
T}@T{
Extend pinkybar with your own crafted scripts written in go
T}
T{
\[en]with\-cpp
T}@T{
\[en]without\-cpp
T}@T{
Extend pinkybar with c++ program
T}
T{
\[en]with\-slang
T}@T{
\[en]without\-slang
T}@T{
Extend pinkybar with slang program
T}
T{
\[en]with\-tcl
T}@T{
\[en]without\-tcl
T}@T{
Extend pinkybar with tcl program
T}
T{
\[en]with\-weather
T}@T{
\[en]without\-weather
T}@T{
The temperature outside (some details must be provided)
T}
T{
api_key=`123458976'
T}@T{
T}@T{
API key obtained after registering yourself in the weather website, must
be combined \f[B]\[en]with\-weather\f[]
T}
T{
\[en]with\-smartemp
T}@T{
\[en]without\-smartemp
T}@T{
Read the drive temperature from S.M.A.R.T cross\-platform available
T}
T{
\[en]with\-drivetemp
T}@T{
\[en]without\-drivetemp
T}@T{
Read the drive temperature from S.M.A.R.T (linux only) uses curl
T}
T{
\[en]with\-drivetemp\-light
T}@T{
\[en]without\-drivetemp\-light
T}@T{
Read the drive temperature from S.M.A.R.T (linux only) light version
T}
T{
drive_port=`1234'
T}@T{
T}@T{
Different TCP port to listen to for the drive temperature, default one
is 7634, must be combined \f[B]\[en]with\-drivetemp\f[] or
\f[B]\[en]with\-drivetemp\-light\f[]
T}
T{
\[en]with\-colours
T}@T{
\[en]without\-colours
T}@T{
Colorize the output data.
T}
T{
icons=/tmp
T}@T{
T}@T{
xbm icons that can be used by dzen2 for example.
Discarded when \f[B]\[en]with\-x11\f[] is used
T}
T{
\[en]with\-mpd
T}@T{
\[en]without\-mpd
T}@T{
To see the currently played song name (if any).
T}
T{
\[en]prefix=/tmp
T}@T{
T}@T{
The directory where the program will be installed
T}
T{
mobo_sensor=`dev.aibs.0'
T}@T{
T}@T{
FreeBSD motherboard sensor module name to use in the sysctl calls.
Read the FreeBSD installation below
T}
T{
cpu_sensor=`dev.cpu.0.temperature'
T}@T{
T}@T{
FreeBSD cpu temperature module name to use in the sysctl calls .
Read the FreeBSD installation below
T}
.TE
.PP
By default, if \f[B]no\f[] options are passed, the program will be
compiled with:
.IP
.nf
\f[C]
\-\-with\-net\ \-\-with\-pci
\f[]
.fi
.PP
Affects only FreeBSD users with laptops, \f[B]\[en]without\-apm\f[] will
compile the program with acpi support to obtain the current battery
life.
.PP
\f[B]\[en]without\-mpd\f[] will compile the program with cmus support,
the options syntax stays as is.
.PP
The pci and sensors configure options will be discarded in *BSD.
.PP
Affects only linux users with wifi/wireless chipsets, run
\f[C]lsmod|grep\ 802\f[] and see whether your chipset uses
cfg80211/mac80211.
If that's so you can rely on libnl and enable \f[B]\[en]with\-libnl\f[]
configure options, otherwise your chipset probably still uses we/wext,
so type \f[B]\[en]without\-libnl\f[].
.PP
Affects only linux users, \f[B]\[en]with\-drivetemp\f[] pretty much
locks you down to hddtemp.
You can adjust \f[B]extra/scripts/drive\-temperature.sh\f[] and compile
the program \f[B]\[en]with\-smartemp\f[], so you can switch between
hddtemp and smartmontools at any time without the need recompile
pinkybar with different options.
\f[B]\[en]with\-smartemp\f[] only cares for the existance of
/tmp/pinkytemp file.
.PP
\f[B]\[en]with\-weather\f[] is using this
url (http://openweathermap.org/current), register yourself there, create
a new API key (https://home.openweathermap.org/api_keys).
.PP
Don't just rush to register yourself, read carefully what the
\[lq]Free\[rq] account limits are and take in account how often the
program should call their api service.
I'm not responsible if you exceeded the limits, you've been warned.
.IP
.nf
\f[C]
#\ Make\ sure\ it\[aq]s\ working\ first
#\ curl\ \[aq]http://api.openweathermap.org/data/2.5/weather?q=London,uk&units=metric&APPID=28459ae16e4b3a7e5628ff21f4907b6f\[aq]

#\ what\ to\ pass\ to\ configure
\-\-with\-weather\ api_key=\[aq]28459ae16e4b3a7e5628ff21f4907b6f\[aq]
\f[]
.fi
.PP
   *   *   *   *   *
.SS Installation for dwm
.IP
.nf
\f[C]
perl\ set.pl\ "distro"
autoreconf\ \-\-install\ \-\-force

\&./configure\ \-\-prefix=$HOME/.cache\ \-\-with\-x11
make
make\ install
\f[]
.fi
.PP
Copy the code from extra/scripts/dwm.sh or \f[C]exec\f[] it from
\f[B]xinitrc\f[] or the script used to start dwm.
.SS Installation for xmonad (or other WM)
.IP
.nf
\f[C]
#\ Copy\ the\ xbm\ icons
mkdir\ \-p\ \-\-mode=700\ $HOME/.xmonad/icons
cp\ \-r\ extra/xbm_icons/*.xbm\ $HOME/.xmonad/icons

perl\ set.pl\ "distro"
autoreconf\ \-\-install\ \-\-force

#\ disable\ X11,\ point\ the\ location\ to\ the\ icons
\&./configure\ \-\-prefix=$HOME/.cache\ \-\-without\-x11\ icons=$HOME/.xmonad/icons

#\ compile\ \[aq]n\ install
make
make\ install
\f[]
.fi
.PP
Copy the code from extra/scripts/xmonad.sh or \f[C]exec\f[] it from
\f[B]xinitrc\f[] or the script used to start xmonad.
.SS Installation in FreeBSD
.PP
FreeBSD has no other way than using the module specific convention to
query sysctl and obtain data from the sensors.
Maintaining a list with all the possible module names and performing
expensive sysctl calls in a loop to determine that X module is loaded
into your system is no\-go.
Be prepared to spend a minute or two to find out some system
information.
.PP
Determine the motherboard sensor module name.
.IP
.nf
\f[C]
sysctl\ \-a|grep\ \[aq]aibs\[aq]

dev.aibs.0.volt.0:\ 1356\ 850\ 1600
dev.aibs.0.volt.1:\ 3344\ 2970\ 3630
dev.aibs.0.volt.2:\ 5040\ 4500\ 5500
dev.aibs.0.volt.3:\ 12278\ 10200\ 13800
dev.aibs.0.temp.0:\ 39.0C\ 60.0C\ 95.0C
dev.aibs.0.temp.1:\ 38.0C\ 45.0C\ 75.0C
dev.aibs.0.fan.0:\ 1053\ 600\ 7200
dev.aibs.0.fan.1:\ 1053\ 600\ 7200
\f[]
.fi
.PP
Copy only `dev.MODULE.NUMBER' (if there is any number at all) and paste
it into the \f[B]mobo_sensor\f[] option below.
.PP
Do the same for your cpu temperature, copy and paste the variable as is.
\f[B]dev.cpu.0.temperature\f[] below is provied as example.
.IP
.nf
\f[C]
perl\ set.pl\ "freebsd"
autoreconf\ \-\-install\ \-\-force

\&./configure\ \-\-prefix=$HOME/.cache\ \-\-with\-x11\ \-\-without\-alsa\ \-\-with\-oss\ mobo_sensor=\[aq]dev.aibs.0\[aq]\ cpu_sensor=\[aq]dev.cpu.0.temperature\[aq]
make
make\ install
\f[]
.fi
.PP
Send a request to the FreeBSD mailing list and request the OpenBSD
sensors API to be ported.
.SS Installation in OpenBSD
.PP
Before proceeding, you'll have to:
.IP
.nf
\f[C]
#\ To\ detect\ the\ newer\ compiler\ that\ you\ are
#\ about\ to\ install
sed\ \-i\ \[aq]s/#AC_PROG_CC(/AC_PROG_CC(/g\[aq]\ configure.ac

ls\ /usr/local/bin/automake\-*
ls\ /usr/local/bin/autoconf\-*

#\ Then\ replace\ the\ numbers\ below
export\ AUTOCONF_VERSION=2.69
export\ AUTOMAKE_VERSION=1.15

#\ Your\ call,\ gcc\ or\ llvm\ ?
pkg_add\ gcc

#\ after\ that:
perl\ set.pl\ "openbsd"
autoreconf\ \-\-install\ \-\-force

\&./configure\ \-\-prefix=$HOME/.cache\ \-\-without\-alsa\ \-\-with\-oss
make
make\ install
\f[]
.fi
.SS pinky curses installation
.PP
Step one, compile pinky\-bar \f[B]\[en]with\-ncurses\f[], so the output
to be formated in a way that pinky\-curses can parse and colorize.
.IP
.nf
\f[C]
perl\ set.pl\ "distro"
autoreconf\ \-\-install\ \-\-force

#\ disable\ X11,\ enable\ the\ colours\ and\ ncurses\ opts.
\&./configure\ \-\-prefix=$HOME/.cache\ \-\-without\-x11\ \-\-with\-alsa\ \-\-with\-colours\ \-\-with\-ncurses

#\ compile\ \[aq]n\ install
make
make\ install
\f[]
.fi
.PP
Step two, compile and install pinky\-curses \-
https://notabug.org/void0/pinky\-curses
.PP
Copy the code from extra/scripts/pinky\-curses.sh
.SS pinky urxvt
.PP
What a coincidence, pinky\-urxvt is my 3rd urxvt extension and 3rd
member of the pinky family.
.PP
The sole purpose of this urxvt extension is to make it easy for you to
keep track of things that you are interested to monitor while hacking
your way something in the terminal.
.PP
Link \- https://notabug.org/void0/pinky\-urxvt
.PP
[IMAGE: image (https://notabug.org/void0/pinky-urxvt/raw/master/2.png)]
.PP
pinky\-urxvt, and pinky\-curses are not tied to pinky\-bar.
.SS Installation for anything else
.PP
pinky\-bar is no longer tied to Window Managers only.
With the addition of \[lq]without colours\[rq], the output can be shown
in any program, just bear in mind that the more options you've supplied
the more system information will be shown.
.PP
The tmux status bar in action:
.PP
[IMAGE: image (img/pic4.png)]
.PP
The installation steps:
.IP
.nf
\f[C]
perl\ set.pl\ "distro"
autoreconf\ \-\-install\ \-\-force

\&./configure\ \-\-prefix=$HOME/.cache\ \-\-without\-x11\ \-\-without\-colours
make
make\ install
\f[]
.fi
.PP
By choosing this 3rd installation method it is up to you where, how to
start and use the system information that's produced by pinky\-bar.
.PP
   *   *   *   *   *
.PP
Replace \f[B]distro\f[] with archlinux, debian, gentoo, slackware, rhel,
frugalware, angstrom.
Here's some short distros list of some popular distros that are based on
another one:
.IP \[bu] 2
[x] archlinux based distros: parabola, chakra, manjaro
.IP \[bu] 2
[x] debian based distros: ubuntu, linux mint, trisquel, back track, kali
linux, peppermint linux, solus, crunchbang, deepin, elementary os, and
the rest *buntu based distros
.IP \[bu] 2
[x] gentoo based distros: funtoo, sabayon, calculate linux
.IP \[bu] 2
[x] slackware
.IP \[bu] 2
[x] rhel based distros: opensuse (uses rpm), fedora, fuduntu, mandriva,
mandrake, viperr, mageia
.IP \[bu] 2
[x] frugalware
.IP \[bu] 2
[x] angstrom
.PP
Cannot list the *BSD flavours as \[lq]distros\[rq], so they deserve own
options:
.IP \[bu] 2
[x] freebsd
.IP \[bu] 2
[x] openbsd
.PP
   *   *   *   *   *
.SS Using configuration file
.PP
\f[B]~/.pinky\f[] is the location of the configuration file.
It uses the same short and long command line options.
.PP
I do advise you to use the long options syntax.
.PP
If any option depends on argument, don't put any space between the
option and the argument.
.PP
Use one option per line.
Contrary to your shell, the \[lq]parser\[rq] won't expand ~/my_script.pl
to point to /home/sweethome/my_script.pl
.IP
.nf
\f[C]
\-\-weather=London,uk
\-\-coresload
\-\-cputemp
\-\-ramperc
\-\-driveperc
\-\-packages
\-\-kernel
\-\-voltage
\-\-fans
\-\-mobo
\-\-mobotemp
\-\-perl=/home/sweethome/my_script.pl
\f[]
.fi
.PP
Execute the program without supplying any command line options and it
will parse the configuration file.
.PP
   *   *   *   *   *
.SS Linux Mandatory requirements
.IP \[bu] 2
gcc/clang
.IP \[bu] 2
glibc
.IP \[bu] 2
autoconf
.IP \[bu] 2
automake
.IP \[bu] 2
m4
.IP \[bu] 2
gawk
.IP \[bu] 2
perl
.IP \[bu] 2
\f[B]as\f[] from binutils
.SS *BSD Mandatory requirements
.IP \[bu] 2
gcc/clang
.IP \[bu] 2
autoconf
.IP \[bu] 2
automake
.IP \[bu] 2
autoconf\-wrapper
.IP \[bu] 2
automake\-wrapper
.IP \[bu] 2
autoconf\-archive
.IP \[bu] 2
argp\-standalone
.IP \[bu] 2
libtool
.IP \[bu] 2
m4
.IP \[bu] 2
gawk
.IP \[bu] 2
perl
.IP \[bu] 2
\f[B]as\f[] from binutils
.PP
Some llvm and gcc versions will not check for headers and libraries in
/usr/local, and you might get something like \f[B]ld cannot find
\-largp\f[], if that's the case for you, you should export the following
environment variables:
.IP
.nf
\f[C]
export\ LDFLAGS=\[aq]\-L/usr/local/lib\[aq]
export\ CFLAGS=\[aq]\-I/usr/local/include\[aq]
\f[]
.fi
.PP
After editing the wrong prototype I managed to stumble upon a bug in
OpenBSD's own libc.
.PP
\f[B]Warning !!! OpenBSD users !!!\f[]
.PP
The majority of SCN* macros differs from their PRI* cousins.
And I cannot guarantee the accuracy of fixed width integers when OpenBSD
own libc managed to use different format specifiers.
Read extra/misc/openbsd_bugs.md for more details.
.SS Opt\-in requirements
.PP
Linux camp:
.PP
The internet related options rely on headers provided iproute2.
By default the program will try to compile with those headers included.
If for any reason you would like to compile the program without internet
related options, then pass \f[B]\[en]without\-net\f[] to configure.
.IP \[bu] 2
iproute2
.PP
wifi/wireless chipsets supporting mac80211/cfg80211:
.IP \[bu] 2
libnl (>= 3.0)
.IP \[bu] 2
pkg\-config
.PP
In Gentoo there are two versions of pkg\-config.
The first one is named dev\-util/pkgconfig and the second one is
dev\-ruby/pkg\-config.
In order to use the first one, you'll have to export the pkg\-config
path to the following environment variable:
.IP
.nf
\f[C]
export\ PKG_CONFIG_PATH=/usr/bin/pkg\-config
\f[]
.fi
.PP
Also you'll have to edit \f[B]m4/extend.m4\f[] and edit the line
\f[C]PKG_CHECK_MODULES([RUBY],\ [ruby\-2.2\ >=\ 2.2]\f[] to point to the
correct ruby version that you have installed.
.PP
Then pass \f[B]\[en]with\-libnl\f[] to configure.
.PP
To get the NIC vendor and model names:
.IP \[bu] 2
pciutils
.PP
Alternative way to obtain data from the sensors:
.IP \[bu] 2
lm_sensors
.PP
To read the drive temperature from S.M.A.R.T
\f[B]\[en]with\-drivetemp\f[]:
.IP \[bu] 2
hddtemp
.IP \[bu] 2
curl
.PP
To read the drive temperature from S.M.A.R.T
\f[B]\[en]with\-drivetemp\-light\f[]:
.IP \[bu] 2
hddtemp
.PP
The \[lq]light\[rq] version does not rely on curl, and will not force
\-O0 CFLAGS.
.IP
.nf
\f[C]
#\ \-\-with\-drivetemp\-light
0.00s\ user\ 0.00s\ system\ 15%\ cpu\ 0.006

#\ \-\-with\-drivetemp
0.01s\ user\ 0.00s\ system\ 72%\ cpu\ 0.008
\f[]
.fi
.PP
Try running hddtemp to see if it detects your drive, depending if it has
temperature sensor in first place:
.IP
.nf
\f[C]
sudo\ hddtemp\ /dev/sda

WARNING:\ Drive\ /dev/sda\ doesn\[aq]t\ appear\ in\ the\ database\ of\ supported\ drives
WARNING:\ But\ using\ a\ common\ value,\ it\ reports\ something.
WARNING:\ Note\ that\ the\ temperature\ shown\ could\ be\ wrong.
WARNING:\ See\ \-\-help,\ \-\-debug\ and\ \-\-drivebase\ options.
WARNING:\ And\ don\[aq]t\ forget\ you\ can\ add\ your\ drive\ to\ hddtemp.db
/dev/sda:\ Corsair\ Force\ GT:\ \ 23°C\ or\ °F
\f[]
.fi
.PP
The message is pretty clear \[lq]don't forget to add your drive to
hddtemp.db\[rq], first run the debug command to see which field is
responsible to report your drive temperature, it should be in the range
of 190 \- 200:
.IP
.nf
\f[C]

extra/py3status/pinkybar.py
# -*- coding: utf-8 -*-
"""
Display system information gathered in pinkybar.

Configuration parameters:
    cache_timeout: refresh interval for this module (default 10)
    command: specify a path to pinkybar command, otherwise auto
        eg '~/.cache/bin/pinkybar' (default None)
    format: display format for this module (default '{output}')
    pythonpath: specify PYTHONPATH to use for scripts (default None)

Format placeholders:
    {output} format for pinkybar output

@author lasers

SAMPLE OUTPUT
pinkybar
[
    {'full_text': u'pinky', 'color': '#ee82ee'},
    {'full_text': u'bar', 'color': '#00bfff'},
]
"""

from os.path import expanduser
STRING_ERROR = 'invalid command'


class Py3status:
    """
    """
    # available configuration parameters
    cache_timeout = 10
    command = None
    format = '{output}'
    pythonpath = None

    def post_config_hook(self):
        if self.command:
            self.command = expanduser(self.command)
        else:
            self.command = self.py3.check_commands(
                ['pinkybar', expanduser('~/.cache/bin/pinkybar')]
            )
        if not self.command or not self.py3.check_commands(self.command):
            raise Exception(STRING_ERROR)
        if self.pythonpath:
            self.command = 'PYTHONPATH={} {}'.format(
                self.pythonpath, self.command
            )
        self.shell = bool(self.pythonpath)

    def pinkybar(self):
        try:
            output = self.py3.command_output(self.command, shell=self.shell)
        except self.py3.CommandError as ce:
            output = ce.output
            if 'pinkybar' in ce.error:
                for line in ce.error.splitlines():
                    if 'pinkybar' in line:
                        self.py3.error(' '.join(line.split()[2:]))

        output = output.strip()

        return {
            'cached_until': self.py3.time_in(self.cache_timeout),
            'full_text': self.py3.safe_format(
                self.format, {'output': self.py3.safe_format(output)}
            )
        }


if __name__ == "__main__":
    """
    Run module in test mode.
    """
    from py3status.module_test import module_test
    module_test(Py3status)
